hooks/react-client/useKeyboardCommands.ts

```diff
@@ -0,0 +1,39 @@
+import { useEffect } from 'react';
+
+type KeyboardCommand = 'cmd-k' | 'escape' | 'down' | 'up' | 'enter';
+
+type KeyboardCommandCallback = (key: KeyboardCommand) => void;
+
+const useKeyboardCommands = (fn: KeyboardCommandCallback) => {
+  useEffect(() => {
+    document.addEventListener('keydown', event => {
+      // Detect ⌘ + k on Mac, Ctrl + k on Windows
+      if ((event.metaKey || event.ctrlKey) && event.key === 'k') {
+        event.preventDefault();
+        fn('cmd-k');
+      }
+
+      if (event.key === 'Escape') {
```

Comment:
This should be a switch statement imo.

Reply:
yeah I have no idea why I wrote it this way. Typical case where I started with just an `if` and escalated quickly